db/columns/operations/create.py

```diff
@@ -1,83 +1,48 @@
-    if column_name == '':
-        column_data[NAME] = gen_col_name(table)
-    column_type_id = column_data.get(TYPE, column_data.get("type"))
+    column_name = (column_data.get(NAME) or '').strip() or None
+    column_type_id = (
+        column_data.get(
+            TYPE, column_data.get("type", PostgresType.CHARACTER_VARYING.id)
+        )
+        or PostgresType.CHARACTER_VARYING.id
+    )
     column_type_options = column_data.get("type_options", {})
     column_nullable = column_data.get(NULLABLE, True)
     default_value = column_data.get(DEFAULT, {}).get('value')
-    prepared_default_value = str(default_value) if default_value is not None else None
-    column_type = get_db_type_enum_from_id(column_type_id)
-    column_type_class = None
-    if column_type is not None:
-        column_type_class = column_type.get_sa_class(engine)
-    if column_type_class is None:
-        # Requested type unknown or not supported. Falling back to CHARACTER_VARYING
-        column_type_class = PostgresType.CHARACTER_VARYING.get_sa_class(engine)
-        column_type_options = {}
-    # TODO reuse metadata
-    table = reflect_table_from_oid(table_oid, engine, metadata=get_empty_metadata())
-
+    col_create_def = [
+        {
+            "name": column_name,
+            "type": {"name": column_type_id, "options": column_type_options},
+            "not_null": not column_nullable,
+            "default": default_value,
+        }
+    ]
     try:
-        column = MathesarColumn(
-            column_data[NAME], column_type_class(**column_type_options), nullable=column_nullable,
-            server_default=prepared_default_value,
+        curr = execute_msar_func_with_engine(
+            engine, 'add_columns',
+            table_oid,
+            json.dumps(col_create_def)
         )
-    except DataError as e:
-        if type(e.orig) == InvalidTextRepresentation:
-            raise InvalidTypeError
-        else:
-            raise e
-
-    # TODO reuse metadata
-    table = reflect_table_from_oid(table_oid, engine, metadata=get_empty_metadata())
-    try:
-        with engine.begin() as conn:
-            ctx = MigrationContext.configure(conn)
-            op = Operations(ctx)
-            op.add_column(table.name, column, schema=table.schema)
-    except DataError as e:
-        if type(e.orig) == InvalidTextRepresentation:
-            raise InvalidDefaultError
-        elif type(e.orig) == InvalidParameterValue:
-            raise InvalidTypeOptionError
-        else:
-            raise e
-
-    # TODO reuse metadata
-    reflected_table = reflect_table_from_oid(table_oid, engine, metadata=get_empty_metadata())
-    reflected_column = reflected_table.columns[column_data[NAME]]
-    return MathesarColumn.from_column(
-        reflected_column,
-        engine=engine
-    )
+    except InvalidTextRepresentation:
+        raise InvalidDefaultError
+    except InvalidParameterValue:
+        raise InvalidTypeOptionError
```

Comment:
I agree. I think we need to do a reworking of our error handling in general. As @silentninja has pointed out, we have error handling happening in inappropriate places in the API and Django, and we're also not doing a great job in the `db` layer. For this PR, I tried to keep the handling as similar to the old version as possible.

I think a longer-term solution would be to have a custom error handling framework in the database that raises far more precise errors for the python layer to pick up. I'd prefer to leave that for a later date, since it's probably a major project.